src/hooks/useTxNotifications.ts

```diff
@@ -33,11 +33,12 @@ const useTxNotifications = (): void => {
         const isSuccess = event === TxEvent.SUCCESS || event === TxEvent.PROPOSED
         const message = isError ? `${baseMessage} ${detail.error.message.slice(0, 300)}` : baseMessage
         const txId = 'txId' in detail && detail.txId
+        const txHash = 'txHash' in detail && detail.txHash
 
         dispatch(
           showNotification({
             message,
-            groupKey: txId || '',
+            groupKey: txHash || txId || '',
```

Comment:
This allows us to group notifications by txHash. Otherwise it would display the same notification multiple times when executing a batch since every transaction has a different txId

Reply:
Makes sense. How about a `batchTxId` then? With the txId of the batch tx.

Reply:
The batch tx is not proposed to the gateway but interacts directly with the multisend contract kind of like a wrapper to execute existing txs so unfortunately there is no txId

Reply:
Then `batchTxHash`. Just to differentiate from the txHash of the tx itself.

Reply:
After looking more into it I realized that we also want to show notifications before there is a txHash e.g. `EXECUTING` so I would propose adding an optional parameter `batch` or `batchId` to some of the events and using that instead.

Reply:
EXECUTING maybe we can remove actually.

Reply:
@iamacook WDYT?

Reply:
Afaik we use `EXECUTING` to show `Please confirm the execution in your wallet.`

Reply:
I refactored the function to `dispatchBatchExecution` which is now almost the same as `dispatchTxExecution` and dispatches the same events but with an additional `batchId`